Improve napalm state output in debug mode

mirceaulinic opened the pull request:
### What does this PR do?

When in debug more, you may want to see what is the rendered config and the diff when available (otherwise you can't actually debug).
Additionally, when there are no changes, the `changes` key should return an empty dict, so the state displays correctly that there were no changes, i.e. `changed=0`.

mirceaulinic committed:
Improve napalm state output in debug mode

salt-jenkins commented:
@mirceaulinic, thanks for your PR! By analyzing the history of the files in this pull request, we identified @bewing and @rallytime to be potential reviewers.

mirceaulinic committed:
Include compliance reports

For the netyang state, the compliance report will be
included by default in the pchanges dictionary.
It can be also returned in the comment field,
by enabling using the `compliance_report` option.

mirceaulinic committed:
Document the new compliance_report arg

mirceaulinic committed:
Simplify the loaded_ret logic

mirceaulinic commented:
I have also simplified the logic under the `loaded_ret` helper function and the display should be a little bit more user-friendly.
I have also pushed a change for the `managed` function, under the `netyang` state to be able to display the compliance report object in the comment of the state output, by adding `compliance_report: true` in the SLS. However, it should always be inside the `pchanges`.

mirceaulinic committed:
Lint and fix

cachedout reviewed (approved):
Really nice!
